### PR TITLE
Add CLIP BetterTransformer

### PR DESCRIPTION
**Pull request by fxmarty**
The support is only for the vision model, as the text model uses causal attention that is likely not supported by PyTorch Better Transformer.

### CODE REVIEW

File: optimum/bettertransformer/transformation.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def transform(
         else:
             load_accelerate = False
 
-        hf_config = model.config
+        hf_config = cfg
```
Comment (younesbelkada) [inline optimum/bettertransformer/transformation.py:155]: Why you need this arg? 🤔

Comment (fxmarty) [reply optimum/bettertransformer/transformation.py:155]: it's draft

File: optimum/bettertransformer/models/encoder_models.py [hunk 2]
```diff
@@ -1095,3 +1101,108 @@ def forward(self, hidden_states, attention_mask, position_bias=None, *_, **__):
...
+        self.validate_bettertransformer()
+
+    def forward(self, hidden_states, attention_mask, *_, **__):
+        r"""
+        This is just a wrapper around the forward function proposed in:
+        https://github.com/huggingface/transformers/pull/19553
+        """
+        super().forward_checker()
+
+        assert attention_mask is None  # we expect attention_mask to be None in the vision model
```
Comment (younesbelkada) [inline optimum/bettertransformer/models/encoder_models.py:1177]:
```suggestion
        # we expect attention_mask to be None in the vision model 
        if attention_mask is not None: 
            raise ValueError("Please do not use attention masks when using `BetterTransformer` converted models") 
```

**Comment by fxmarty** [comment]
Tests pass locally on GPU, so I'll merge. I don't think there is any workflow for GPU tests?

**Pull request merged**